src/transformers/modeling_utils.py

```diff
@@ -864,6 +924,11 @@ def save_pretrained(
         # Only save the model itself if we are using distributed training
         model_to_save = unwrap_model(self)
 
+        # save the string version of dtype to the config, e.g. convert torch.float32 => "float32"
+        # so in from_config and from_pretrained we reverse this with getattr(torch, "float32")
+        dtype = get_parameter_dtype(model_to_save)
+        model_to_save.config.torch_dtype = str(dtype).split(".")[1]
```

Comment:
This should be modified to do so only under pytorch. How do we know inside `save_pretrained` if it's a pytorch model?

Reply:
If you are here, you are a PyTorch model. TensorFlow models are in `TFPreTrainedModel.save_pretrained` and same for Flax.